hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/cache/EpochEntry.java

```diff
@@ -64,6 +64,9 @@ public int hashCode() {
   @Override
   public int compareTo(Object o) {
     if (this.epoch == ((EpochEntry<?>)o).epoch) {
+      if (this.cachekey != ((EpochEntry<?>)o).cachekey) {
+        return -1;
```

Comment:
> The implementor must ensure sgn(x.compareTo(y)) == -sgn(y.compareTo(x)) for all x and y.

In other words, we cannot return -1 unconditionally in this case.

Reply:
> > The implementor must ensure sgn(x.compareTo(y)) == -sgn(y.compareTo(x)) for all x and y.
> 
> In other words, we cannot return -1 unconditionally in this case.

yep, I use hashcode to compare now.